router/router.go

```diff
@@ -2213,31 +2212,19 @@ func (rt *HandleT) Start() {
 	rt.logger.Infof("Starting router : %s", rt.destName)
 	ctx := rt.backgroundCtx
 	rt.backgroundGroup.Go(func() error {
-		select {
-		case <-ctx.Done():
-			rt.logger.Infof("Router : %s start goroutine exited", rt.destName)
-			return nil
-		case <-rt.backendConfigInitialized:
-			// no-op, just wait
-		}
+		<-rt.backendConfigInitialized
```

Comment:
Why did we remove the select between `ctx.Done()` and `rt.backendConfigInitialized` are we sure router will Shutdown if backendconfig is not initialized?

Reply:
`Start` performs its operations asynchronously, through a go routine, effectively returning before it is actually done. So the error we are fixing here occurred because we were calling `Stop` before the router was actually started, right?

Reply:
Without the generator loop starting the shutdown would not happen because for the shutdown to happen successfully we would need to close the worker channels. If we have `ctx.Done()` here and the `stop` function is called before the generator loop gets to start we would have the router in a stuck state which was happening in the tests

Reply:
The context is just the group context and this routine is part of the group so it doesn't make sense to wait for this context specifically, right?